exodus_lambda/functions/origin_request.py

```diff
@@ -165,18 +165,22 @@ def handler(self, event, context):
         )
 
         if query_result["Items"]:
-            self.logger.info("Item found for '%s'", uri)
-
             try:
+                # Validate If the item's "object_key" is "absent"
+                object_key = query_result["Items"][0]["object_key"]["S"]
+                if object_key == "absent":
+                    self.logger.info("Item absent for '%s'", uri)
+                    return {"status": "404", "statusDescription": "Not Found"}
+                else:
```

Comment:
Can the "else" please be dropped to reduce unnecessary nesting?

It has no effect since the "if" case always returns.  It's also arbitrary that you put only one line of code in it.  Every line of code between here and "return request" should execute only in the "item found" case, what's special about this one line of code which makes sense for it to be in "else" and not any of the others?